src/bin/stratisd.rs

```diff
@@ -109,8 +134,13 @@ fn run() -> StratisResult<()> {
             }
         }
 
-        // Ask the engine to check its pools
-        engine.borrow_mut().check()
+        // Refresh list of dbus fds to poll for every time. This can change as
+        // D-Bus clients come and go.
+        fds.truncate(dbus_fd_start_idx);
+        fds.extend(dbus_conn.watch_fds().iter().map(|w| w.to_pollfd()));
+
+        let r = unsafe { libc::poll(fds.as_mut_ptr(), fds.len() as libc::c_ulong, poll_timeout) };
+        assert!(r >= 0);
```

Comment:
I think ```assert!``` is probably not the write way to check ```poll()``` return code.

Reply:
We really shouldn't be using `libc::poll` at all. I've opened #655, which will include removing that assert.